docker/client.go

```diff
@@ -86,16 +137,54 @@ func ListContainers(socketPath string) ([]Container, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to create Docker client: %v", err)
 	}
+
 	defer cli.Close()
 
+	hostContainer, err := getHostContainer(ctx, cli)
+	if enforceNetworkValidation && err != nil {
+		return nil, fmt.Errorf("network validation enforced, cannot validate due to: %v", err)
```

Comment:
should use `%w` here instead of `%v` for error so that the error populate up the stack and get printed with all the stack trace